src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -52,342 +56,3008 @@ const ActorInit Boss_Ganon2_InitVars = {
-        { 15, { { 0, 0, 0 }, 30 }, 100 },
-    },
-};
+#include "z_boss_ganon2_data.c"
 
-static ColliderJntSphInit D_80906FBC = {
-    {
-        COLTYPE_METAL,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_PLAYER,
-        OC2_FIRST_ONLY | OC2_TYPE_1,
-        COLSHAPE_JNTSPH,
-    },
-    16,
-    D_80906D7C,
-};
+void BossGanon2_InitRand(s32 seedInit0, s32 seedInit1, s32 seedInit2) {
+    sSeed1 = seedInit0;
+    sSeed2 = seedInit1;
+    sSeed3 = seedInit2;
+}
 
-static ColliderJntSphElementInit D_80906FCC[2] = {
-    {
-        {
-            ELEMTYPE_UNK2,
-            { 0xFFCFFFFF, 0x00, 0x40 },
-            { 0xFFDFFFFF, 0x00, 0x00 },
-            TOUCH_ON | TOUCH_SFX_NORMAL,
-            BUMP_ON,
-            OCELEM_ON,
-        },
-        { 15, { { 0, 0, 0 }, 45 }, 100 },
-    },
-    {
-        {
-            ELEMTYPE_UNK2,
-            { 0xFFCFFFFF, 0x00, 0x40 },
-            { 0xFFDFFFFF, 0x00, 0x00 },
-            TOUCH_ON | TOUCH_SFX_NORMAL,
-            BUMP_ON,
-            OCELEM_ON,
-        },
-        { 16, { { 0, 0, 0 }, 45 }, 100 },
-    },
-};
+f32 BossGanon2_RandZeroOne(void) {
+    // Wichmann-Hill algorithm
+    f32 randFloat;
 
-static ColliderJntSphInit D_80907014 = {
-    {
-        COLTYPE_METAL,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_PLAYER,
-        OC2_TYPE_1,
-        COLSHAPE_JNTSPH,
-    },
-    2,
-    D_80906FCC,
-};
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_808FCF40.s")
+    sSeed1 = (sSeed1 * 171) % 30269;
+    sSeed2 = (sSeed2 * 172) % 30307;
+    sSeed3 = (sSeed3 * 170) % 30323;
+
+    randFloat = (sSeed1 / 30269.0f) + (sSeed2 / 30307.0f) + (sSeed3 / 30323.0f);
+    while (randFloat >= 1.0f) {
+        randFloat -= 1.0f;
+    }
+    return fabsf(randFloat);
+}
+
+void func_808FD080(s32 idx, ColliderJntSph* collider, Vec3f* arg2) {
+    collider->elements[idx].dim.worldSphere.center.x = arg2->x;
+    collider->elements[idx].dim.worldSphere.center.y = arg2->y;
+    collider->elements[idx].dim.worldSphere.center.z = arg2->z;
+
+    collider->elements[idx].dim.worldSphere.radius =
+        collider->elements[idx].dim.modelSphere.radius * collider->elements[idx].dim.scale;
+}
+
+void BossGanon2_SetObjectSegment(BossGanon2* this, GlobalContext* globalCtx, s32 objectId, u8 arg3) {
+    s32 pad;
+    s32 objectIdx = Object_GetIndex(&globalCtx->objectCtx, objectId);
+
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[objectIdx].segment);
+
+    if (arg3) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 790);
+
+        gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[objectIdx].segment);
+        gSPSegment(POLY_XLU_DISP++, 0x06, globalCtx->objectCtx.status[objectIdx].segment);
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 799);
+    }
+}
+
+void func_808FD210(GlobalContext* globalCtx, Vec3f* arg1) {
```

Comment:
BossGanon2_SetEffectPosition ?

Reply:
this is doing more than setting a positioin

Reply:
I guess I didn't read the whole function? Thought I did